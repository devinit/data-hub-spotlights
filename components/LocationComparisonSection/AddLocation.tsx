import React, { FunctionComponent } from 'react';

interface AddLocationProps {
  label: string;
}

const AddLocation: FunctionComponent<AddLocationProps> = ({ label }) => {
  return (
    <a href="#" className="m-text-link">
      <i role="presentation" aria-hidden="true" className="ico ico--16 ico-plus-poppy"></i>
      <span>{label}</span>
    </a>
  );
};

export { AddLocation };
