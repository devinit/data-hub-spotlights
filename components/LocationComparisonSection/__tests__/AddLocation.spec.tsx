/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { AddLocation } from '../AddLocation';

const onWidgetClick = jest.fn();

describe('Add Location', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <AddLocation active={true} label={'Add Location'} onWidgetClick={onWidgetClick} />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('It does not render when active class is false', () => {
    const renderer = TestRenderer.create(
      <AddLocation active={false} label={'Add Location'} onWidgetClick={onWidgetClick} />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
