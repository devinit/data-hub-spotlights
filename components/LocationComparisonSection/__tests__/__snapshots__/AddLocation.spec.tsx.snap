// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Location It does not render when active class is false 1`] = `
<a
  className="m-text-link add-location-link"
  onClick={[Function]}
>
  <style
    jsx={true}
  >
    
        .add-location-link {
          display: none;
          cursor: pointer;
        }
      
  </style>
  <i
    aria-hidden="true"
    className="ico ico--16 ico-plus-poppy"
    role="presentation"
  />
  <span>
    Add Location
  </span>
</a>
`;

exports[`Add Location renders correctly 1`] = `
<a
  className="m-text-link add-location-link"
  onClick={[Function]}
>
  <style
    jsx={true}
  >
    
        .add-location-link {
          display: block;
          cursor: pointer;
        }
      
  </style>
  <i
    aria-hidden="true"
    className="ico ico--16 ico-plus-poppy"
    role="presentation"
  />
  <span>
    Add Location
  </span>
</a>
`;
